File: Search Insert Position.cpp
#define F(i, a, b) for(int i = a; i <= b; i++)
#include <vector>

class Solution {
public:
    int searchInsert(int A[], int n, int target) {
        int begin = 0;
        int end = n-1;
        int mid = (begin + end)/2;

        while( mid != begin ){
            if( target == A[mid] ){
                return mid;
            }
            if( target < A[mid] ){
                end = mid;
            }else{
                begin = mid;
            }
            mid = (begin + end)/2;
        }

        while( (mid + 1 <= n) && target > A[mid] ){
            mid++;
        }
        while( (mid - 1 >= 0) && target < A[mid-1] ){
            mid--;
        }

        return mid;
    }
};

File: LeetCode-Java/Search Insert Position.java




public class Solution {
	public int searchInsert(int[] A, int target) {
		int len = A.length;

		if( target > A[len-1] ){
			return len; 
		}

		for( int i = 0; i <= len-1; i++ ){
			if( target <= A[i] ){
				return i;
			}
		}

		return -1;
	}
}



